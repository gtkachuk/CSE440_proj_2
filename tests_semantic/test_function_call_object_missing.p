program testFunctionCallObjectMissing;

class myCustomClass

BEGIN
   
   VAR
      lightSwitch	      : boolean;
END



class testFunctionCallObjectMissing

BEGIN
   
   VAR
      onOff	     : boolean;
      compilerWorks  : boolean;   
      myCustomObject : myCustomClass;

FUNCTION setCompilerWorks(value	: boolean ): integer;
BEGIN
   compilerWorks := value;
   setCompilerWorks := 9
END;   

FUNCTION testFunctionCallObjectMissing;
BEGIN
      onOff := myCustomObject.setCompilerWorks(onOff)
END

END
.
