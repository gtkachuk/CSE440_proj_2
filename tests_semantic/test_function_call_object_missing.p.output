ERROR:31:Type mismatch between "boolean" and "myCustomClass"
Errors detected. Exiting.
