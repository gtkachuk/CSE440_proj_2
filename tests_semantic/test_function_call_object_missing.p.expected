ERROR:31:Function "setCompilerWorks" not declared
Errors detected. Exiting.
